x/storage/keeper/keeper.go

```diff
@@ -377,6 +379,14 @@ func (k Keeper) UpdateBucketInfo(ctx sdk.Context, operator sdk.AccAddress, bucke
 		return types.ErrSourceTypeMismatch
 	}
 
+	//TODO rename the  harfork
+	if ctx.IsUpgraded(upgradetypes.Eddystone) {
```

Comment:
Why need this check? 1) What about sp is maintainence status? 2) If a user only wants to update payment account?

Reply:
This is mainly for restricting user to update the quota, to avoid the loss. In the simplified SP exit process, the quota info on exit SP will not 'migrated' to successor.